extensions/grafana-loki/PushGrafanaLokiREST.h

```diff
@@ -0,0 +1,178 @@
+    .withDescription("The tenant ID used by default to push logs to Grafana Loki. If omitted or empty it assumes Grafana Loki is running in single-tenant mode and no X-Scope-OrgID header is sent.")
+    .build();
+  EXTENSIONAPI static constexpr auto MaxBatchSize = core::PropertyDefinitionBuilder<>::createProperty("Max Batch Size")
+    .withDescription("The maximum number of flow files to process at a time. If not set, or set to 0, all FlowFiles will be processed at once.")
+    .withPropertyType(core::StandardPropertyTypes::UNSIGNED_LONG_TYPE)
+    .withDefaultValue("100")
+    .build();
+  EXTENSIONAPI static constexpr auto LogLineBatchWait = core::PropertyDefinitionBuilder<>::createProperty("Log Line Batch Wait")
+    .withDescription("Time to wait before sending a log line batch to Grafana Loki, full or not. If this property and Log Line Batch Size are both unset, "
+                     "the log batch of the current trigger will be sent immediately.")
+    .withPropertyType(core::StandardPropertyTypes::TIME_PERIOD_TYPE)
+    .build();
+  EXTENSIONAPI static constexpr auto LogLineBatchSize = core::PropertyDefinitionBuilder<>::createProperty("Log Line Batch Size")
+    .withDescription("Number of log lines to send in a batch to Loki. If this property and Log Line Batch Wait are both unset, "
+                     "the log batch of the current trigger will be sent immediately.")
+    .withPropertyType(core::StandardPropertyTypes::UNSIGNED_INT_TYPE)
+    .build();
+  EXTENSIONAPI static constexpr auto ConnectTimeout = core::PropertyDefinitionBuilder<>::createProperty("Connection Timeout")
+    .withDescription("Max wait time for connection to the Grafana Loki service.")
+    .withPropertyType(core::StandardPropertyTypes::TIME_PERIOD_TYPE)
+    .withDefaultValue("5 s")
+    .isRequired(true)
+    .build();
+  EXTENSIONAPI static constexpr auto ReadTimeout = core::PropertyDefinitionBuilder<>::createProperty("Read Timeout")
+    .withDescription("Max wait time for response from remote service.")
+    .withPropertyType(core::StandardPropertyTypes::TIME_PERIOD_TYPE)
+    .withDefaultValue("15 s")
+    .isRequired(true)
+    .build();
+  EXTENSIONAPI static constexpr auto SSLContextService = core::PropertyDefinitionBuilder<>::createProperty("SSL Context Service")
+    .withDescription("The SSL Context Service used to provide client certificate information for TLS/SSL (https) connections.")
+    .withAllowedTypes<minifi::controllers::SSLContextService>()
+    .build();
+  EXTENSIONAPI static constexpr auto Username = core::PropertyDefinitionBuilder<>::createProperty("Username")
+    .withDescription("Username for authenticating using basic authentication.")
+    .build();
+  EXTENSIONAPI static constexpr auto Password = core::PropertyDefinitionBuilder<>::createProperty("Password")
+    .withDescription("Password for authenticating using basic authentication.")
+    .build();
```

Comment:
as the sensitive properties have been merged, should we mark this as sensitive?

Reply:
Good catch, made it sensitive in 5d54d81073b42bd82c75788792e6795bf12a8448

Reply:
the description in PROCESSORS.md should also reflect this change

Reply:
You're right I missed that, added in 54448c7c7ea0e645eea0f1e91ad39e715e3ba0a5